### PR TITLE
Add a README.rst

### PR DESCRIPTION
**Pull request by smasoka**
Updated README and HISTORY
I also did some minor changes on cache, interface and downloads. 
I added a real test that downloads 50MB from google-drive but marked to be skipped so it doesn't run unless you are sure since it uses data.

### CODE REVIEW

File: HISTORY.rst [hunk 1]
```diff
@@ -4,6 +4,7 @@ History
 
 X.Y.Z (YYYY-MM-DD)
 ------------------
+* Add README file (:pr: `6`) and minor fixes
```
Comment (sjperkins) [inline HISTORY.rst:7]: I think the PR is number #11?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
+# packratt
```
Comment (sjperkins) [inline README.md:1]: Let's remove the markdown version and keep the restructured text version.

File: README.rst [hunk 3]
```diff
@@ -0,0 +1,73 @@
+packratt
```
Comment (sjperkins) [inline README.rst:1]: Looks good!

File: packratt/cache.py [hunk 4]
```diff
@@ -90,6 +90,9 @@ def __setitem__(self, key, value):
         with open(entry_dir / "entry.yaml", "w") as f:
             f.write(yaml.safe_dump(value))
 
+    def set_cache_entry(self, key, entry):
```
Comment (sjperkins) [inline packratt/cache.py:93]: Do we need this? Can we not just use 

```
cache[key] = value
```

which invokes ``Cache.__setitem__``?

File: packratt/cache.py [hunk 5]
```diff
@@ -127,6 +130,9 @@ def __getitem__(self, key):
 
         return entry
 
+    def get_cache_entry(self, key):
```
Comment (sjperkins) [inline packratt/cache.py:133]: Do we need this? Can we not just use?


```python
value = cache[key]
```

which invokes ``Cache.__getitem__``?

File: HISTORY.rst (hunk 1 above)
Comment (smasoka) [reply HISTORY.rst:7]: not 6, 11 yes

**Pull request merged**